src/status_im2/contexts/emoji_picker/view.cljs

```diff
@@ -0,0 +1,200 @@
+(ns status-im2.contexts.emoji-picker.view
+  (:require [oops.core :as oops]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as quo.theme]
+            [react-native.blur :as blur]
+            [react-native.core :as rn]
+            [react-native.gesture :as gesture]
+            [react-native.platform :as platform]
+            [reagent.core :as reagent]
+            [status-im2.contexts.emoji-picker.constants :as constants]
+            [status-im2.contexts.emoji-picker.data :as emoji-picker.data]
+            [status-im2.contexts.emoji-picker.style :as style]
+            [status-im2.contexts.emoji-picker.utils :as emoji-picker.utils]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [utils.transforms :as transforms]))
+
+(defn- on-press-category
+  [id index active-category scroll-ref]
+  (reset! active-category id)
+  (some-> ^js @scroll-ref
+          (.scrollToIndex #js
+                           {:index    (emoji-picker.data/get-section-header-index-in-data index)
+                            :animated false})))
+
+(defn- handle-on-viewable-items-changed
+  [event active-category filtered-data]
+  (when-not @filtered-data
```

Comment:
I think it's better whenever we can to not pass atoms around to help keep functions less stateful and/or hopefully pure. `filtered-data` can be passed deref'ed already.

`(empty? @filtered-data)` is more idiomatic to check if there's nothing in the sequence. `(empty? nil)` is also valid, although as I mentioned in another comment, I think it's often beneficial to initialize vectors as `[]` instead of `nil`.

Reply:
The filtered data is used to check whether the search data is displayed are not as we don't have to check or update the `active-category` while displaying search results.